ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -185,10 +218,17 @@ StatusDialog {
 
         ErrorTag {
             objectName: "errorTag"
-            visible: root.swapAdaptor.swapOutputData.hasError
+            visible: root.swapAdaptor.swapOutputData.hasError || payPanel.amountEnteredGreaterThanBalance
             Layout.alignment: Qt.AlignHCenter
             Layout.topMargin: Style.current.smallPadding
-            text: qsTr("An error has occured, please try again")
+            text: {
+                if (payPanel.amountEnteredGreaterThanBalance)   {
+                    return qsTr("Insufficient funds for swap")
+                }
+                return qsTr("An error has occured, please try again")
+            }
+            buttonText: payPanel.amountEnteredGreaterThanBalance ? qsTr("Buy crypto"): ""
+            onButtonClicked: console.warn("TODO::: add buy crypto flow")
```

Comment:
Don't we have already a "Buy crypto" popup?

```suggestion
            onButtonClicked: Global.openBuyCryptoModalRequested()
```

We do :)

Reply:
good point, done!